src/components/store/store.tsx

```diff
@@ -0,0 +1,194 @@
+import React, {
+  Dispatch,
+  SetStateAction,
+  createContext,
+  useCallback,
+  useContext,
+  useState,
+} from 'react';
+
+import { Todo } from '../../types/Todo';
+import { ProviderProps } from '../../types/ProviderProps';
+import { FilterOption } from '../../enum/FilterOption';
+import { filterTodos } from '../../helpers/filterTodos';
+import { ErrorOption } from '../../enum/ErrorOption';
+import { apiClient } from '../../api/todos';
+
+type TodosContextType = {
+  todos: Todo[],
+  hasError: ErrorOption | null;
+  resetHasError: () => void;
+  setError: (errMessage: ErrorOption) => void;
+  addTodo: (newTodo: Todo) => void;
+  removeTodo: (todoId: number) => void;
+  recieveTodos: (newTodos: Todo[]) => void,
+  setFilterSelected: (filter: FilterOption) => void,
+  filterSelected: FilterOption,
+  filteredTodos: Todo[],
+  toggleTodoCondition: (
+    todo: Todo,
+    setIsLoading: Dispatch<SetStateAction<boolean>>,
+  ) => void,
+  updateTodoTitle: (
+    todoId: number,
+    title: string,
+  ) => void,
+  resetDeletingTodoIds: () => void,
+  addDeletingTodoIds: (newTodoIds: number[]) => void,
+  deletingTodoIds: number[],
+  toggleAllTodoCondition: () => void,
+};
+
+const TodosContext = createContext<TodosContextType>({
+  todos: [],
+  hasError: ErrorOption.AddTodoError,
+  resetHasError: () => {},
+  setError: () => {},
+  addTodo: () => {},
+  removeTodo: () => {},
+  recieveTodos: () => {},
+  setFilterSelected: () => {},
+  filterSelected: FilterOption.All,
+  filteredTodos: [],
+  toggleTodoCondition: () => {},
+  resetDeletingTodoIds: () => {},
+  addDeletingTodoIds: () => {},
+  deletingTodoIds: [],
+  toggleAllTodoCondition: () => {},
+  updateTodoTitle: () => {},
+});
+
+export const TodosContextProvider: React.FC<ProviderProps> = ({
+  children,
+}) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [hasError, setHasError] = useState<ErrorOption | null>(null);
+  const [deletingTodoIds, setDeletingTodoIds] = useState<number[]>([]);
```

Comment:
It should be not only deleting todo ids, but all loading todo ids. For example when a user clicks toggle all or changes/deleting todo you should add ids too.
As for now I don't see a loader when toggle all and can't understand what todos currently updating